react/react1/week3/react1week3-shifts/src/components/InputShift.js

```diff
@@ -0,0 +1,98 @@
+import React, { useState } from 'react';
+import { ViewShiftList } from './ViewShiftList';
+import { SearchShift } from './SearchShift';
+
+import { DataDefaultShiftApi } from './DataDefaultShiftApi';
+
+export function InputShift() {
+  const [inputShift, setInputShift] = useState([]);
+
+  const [name, setName] = useState('');
+  const [start, setStartTime] = useState('');
+  const [end, setEndTime] = useState('');
+  const [searchName, setSearchName] = useState('');
+  const [dataFechingState, setDataFechingState] = useState(false);
+
+  function addShift() {
+    const startEntry = start.split(':');
+    const startMinutes = Number(startEntry[0]) * 60 + Number(startEntry[1]);
+    const endEntry = end.split(':');
+    const endMinutes = Number(endEntry[0]) * 60 + Number(endEntry[1]);
+    const totalMinutes = endMinutes - startMinutes;
+    const hours = Math.floor(totalMinutes / 60);
+    const minutes = totalMinutes % 60;
+
+    if (hours > 0 || minutes > 0) {
+      const addedShift = [...inputShift, { name, start, end }];
+      setInputShift(addedShift);
+      alert('shift saved');
+    } else {
+      setInputShift(inputShift);
+      alert('check the time');
+    }
+  }
+
+  function deleteShift(name) {
+    setInputShift((shifts) => {
+      const shiftsAfterDelete = shifts.filter((shift) => shift.name !== name);
+      return shiftsAfterDelete;
+    });
+  }
+
+  const shifts = inputShift
+    .filter((val) => val.name.toLowerCase().includes(searchName.toLowerCase()))
+    .map((item) => {
+      return (
+        <ViewShiftList
+          name={item.name}
+          start={item.start}
+          end={item.end}
+          key={item.name}
+          deleteShift={deleteShift}
```

Comment:
Key with name is not a good option. What happen if same person save himself again. So program will give some crash with filter. You can use index for key. It will give item index automatically from array. 
```suggestion
    .map((item, index) => {
      return (
        <ViewShiftList
          name={item.name}
          start={item.start}
          end={item.end}
          key={item.index}
          deleteShift={deleteShift}
```

Reply:
i changed it but it says (in line 44)  it says declared but its value never used !??